Create assignment1_problems1-3.md

castrorazo opened the pull request:
Working on problem questions for assignment one

castrorazo committed:
Create assignment1_problems1-3.md

Working on problem questions for assignment one

castrorazo committed:
Update assignment1_problems1-3.md

Removed some of the old text code, and answered problem 1.

castrorazo committed:
Update README.md

Removed my answers from the README file that I added by accident.

castrorazo committed:
Update assignment1_problems1-3.md

Adding to my question one.

castrorazo committed:
Update assignment1_problems1-3.md

castrorazo committed:
Update assignment1_problems1-3.md

castrorazo committed:
Update assignment1_problems1-3.md

castrorazo committed:
Update assignment1_problems1-3.md

castrorazo committed:
Update assignment1_problems1-3.md

castrorazo committed:
Update assignment1_problems1-3.md

castrorazo committed:
Create normal.txt

castrorazo committed:
Added Source Code to Repo for Assignment One

This code should contain the updated code used to display the UART message done from example code with my name in the string section.

castrorazo committed:
Delete Hello-World-Source/normal.txt

Was not going to use this random txt file.

castrorazo committed:
Create STM-Tera-Output

castrorazo committed:
Update and rename STM-Tera-Output to STM-Tera-Output.md

Added the UART output console message image in the repository as the third deliverable item.

castrorazo committed:
Update README.md

castrorazo committed:
Update assignment1_problems1-3.md

Finalized all my problems.

castrorazo committed:
Create nomal.txt

just a folder made for source code

castrorazo committed:
Add files via upload

castrorazo committed:
Create normal.md

castrorazo committed:
Core Folder Files

castrorazo committed:
Delete Hello-World-Source directory

Redundant Information

castrorazo committed:
Delete Assignments/Assignment01/Core/Src/main.c

Removed old main.c file to be able to add new main.c file with student name.

castrorazo committed:
main.c file update

glenn-andrews commented on assignment1_problems1-3.md:
```diff
@@ -0,0 +1,42 @@
+-   [x] What is an embedded system you would like to investigate?
+ * One fascinating embedded system to me is energy management system. One that includes the incoming grid power, a generator, an ESS, solar panels, and discharges to your electric car.
+-   [x] Describe how you think it works
+ * Rudamentary systems with just a backup generator need a basic ATS system. However when you have mutiple incoming sources, and potential high power loads this isn't the case. You need some type of MID, and that device requires a lot of embedded systems. 
```
I'm not familiar with the acronym, what is an MID?

glenn-andrews commented on assignment1_problems1-3.md:
```diff
@@ -0,0 +1,42 @@
+
+-   [x] Why are bitwise operations so important?
+    - Bitwise operations are important because they provide versatile tools for working with individual bits of data, which is essential in various aspects of embedded systems programming.
+        - One of the tools it helps bring to light is binary formatting and manipulation. Datasheets often show the information we need in other number systems and it's essential to be able to work with this data.
```
I'm not sure what you mean by this. Please explain.

glenn-andrews commented on assignment1_problems1-3.md:
```diff
@@ -0,0 +1,42 @@
+### Problem 3
+
+-   [x] Why are bitwise operations so important?
+    - Bitwise operations are important because they provide versatile tools for working with individual bits of data, which is essential in various aspects of embedded systems programming.
```
Please expand on this. HOW are they essential was the question...

glenn-andrews commented on assignment1_problems1-3.md:
```diff
@@ -0,0 +1,42 @@
+-   [x] Why are bitwise operations so important?
+    - Bitwise operations are important because they provide versatile tools for working with individual bits of data, which is essential in various aspects of embedded systems programming.
+        - One of the tools it helps bring to light is binary formatting and manipulation. Datasheets often show the information we need in other number systems and it's essential to be able to work with this data.
+        - Another tool that bitwise operations let you use is create bit level algorithms such as bitmaps. Bitwise operations are as fundamental to bitmaps as algebra is to calculus. 
```
Explain why they are fundamental. This reads like a line from a textbook or ChatGPT with no understanding behind it

glenn-andrews commented on README.md:
```diff
@@ -1,3 +1,4 @@
+# ![embedded-real-time-systems-2000x600](https://github.com/castrorazo/EMBSYS310-assignment01-/assets/147465177/42261c57-9e5e-4c19-84ca-3ca0b2f73b5f)
```
Back out the changes to this file

glenn-andrews reviewed (changes requested):
Not entirely sure you've got a handle on question 3

castrorazo committed:
Update assignment1_problems1-3.md

Changes from the comments of Professor Glenn.

castrorazo committed:
Update assignment1_problems1-3.md

Updated Question 1, and Question 3.

glenn-andrews commented on assignment1_problems1-3.md:
```diff
@@ -0,0 +1,40 @@
+
+
+
+## Create Pull Request
```
I don't think this belongs here.

glenn-andrews reviewed (approved):
Close enough, but in the future please watch out for modifying files that don't need to be modified for the purposes of the assignment.

glenn-andrews reviewed (changes requested):
Screenshot does not appear in the .md file. Can you verify it's there?

castrorazo commented:
![image](https://github.com/castrorazo/EMBSYS310-assignment01-/assets/147465177/c7991453-51b2-4cc5-893b-6cf89428945f)
Hello, I have the screenshot as a markdown file "STM-Tera-Output.md". Please let me know if you don't see this file on your end. I can also try including it in my source code folder as another md file if that helps?
![image](https://github.com/castrorazo/EMBSYS310-assignment01-/assets/147465177/6353d3ef-fce2-4e8c-ac1e-eed78bd42957)

castrorazo committed:
Delete Assignments/Assignment01/Core/Src/main.c

Removing my modified main.c file, and will be adding in the previous main.c provided by professor.

castrorazo committed:
Replacing the main.c file with the orignal

glenn-andrews reviewed (changes requested):
Why did you back out the changes required for the assignment?

I'm REALLY confused by what you're doing here.

castrorazo commented:
The changes I just made reversed the main.c file to its original state per your request.
![image](https://github.com/castrorazo/EMBSYS310-assignment01-/assets/147465177/c2274494-b52a-4ac5-ac74-b54d7f0c49ff)

glenn-andrews commented:
> The changes I just made reversed the main.c file to its original state per your request. ![image](https://user-images.githubusercontent.com/147465177/275620016-c2274494-b52a-4ac5-ac74-b54d7f0c49ff.png)

The files I meant was the `README.md` and copying part of it to the solutions file.

Changing `main.c` is part of the assignment.

castrorazo commented:
Oh I see, I shouldn't be including anything other than the solutions in that file. Sorry about the confusion. Won't happen again.

castrorazo committed:
Update assignment1_problems1-3.md - 

Removed the extra text not relevant to the solution file

castrorazo committed:
Update assignment01_problems1-3.md

Putting my answers in the provided document for solutions. Didn't even see you put these here for us. Sorry.

castrorazo committed:
Update assignment01_term_output.md

Adding in my screenshot from the Tera Term output display. My name is Richard Castro. I tried to leave the rest of the source code in tact, sorry for changing other things.

castrorazo committed:
Delete STM-Tera-Output.md

Removing this file since it's a repeat of the file provided by the professor.

castrorazo committed:
Delete assignment1_problems1-3.md

Removing this repeated file since it's a duplicate of the file already provided by the professor to add in my solutions to the problems.

castrorazo commented:
Hey Professor, I made sure to follow the instructions for how to answer the questions, and put things where they were meant to be. I appreciate you answering my questions about this during lecture, I was a bit confused when trying to do this.
